src/components/notes/new-route.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect } from 'react'
+import { useClient } from 'cozy-client'
+import { createNoteDocument, generateReturnUrlToNotesIndex } from 'lib/utils'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+import get from 'lodash/get'
+import { generateWebLink, useCapabilities } from 'cozy-client'
+
+const NewRoute = () => {
+  const client = useClient()
+  const { capabilities } = useCapabilities(client)
+  const isFlatDomain = get(capabilities, 'flat_subdomains')
+
+  useEffect(() => {
+    const createNote = async () => {
+      const { data: doc } = await createNoteDocument(client)
+      const returnUrl = generateWebLink({
+        slug: 'notes',
+        cozyUrl: client.getStackClient().uri,
+        subDomainType: isFlatDomain ? 'flat' : 'nested',
+        pathname: `/`
+      })
+      window.location.href = await generateReturnUrlToNotesIndex(
```

Comment:
Hmm, I think we should not directly change the href. 

Let's do the same we it's already done in Drive: https://github.com/cozy/cozy-drive/blob/953e2c4e25e0c8461b9813f18599d9aa39d38bd2/src/drive/web/modules/drive/Toolbar/components/CreateNoteItem.jsx#L75-L85